src/support.cpp

```diff
@@ -906,14 +906,30 @@ void AreaSupport::generateSupportAreasForMesh(SliceDataStorage& storage,
                                                    // area_beyond_limit is the portion of the layer below's outline that lies further away from the current layer's outline than limit_distance
                                                    Polygons area_beyond_limit = mesh.layers[layer_idx - 1].getOutlines().difference(mesh.layers[layer_idx].getOutlines().offset(limit_distance));
 
-                                                   if (area_beyond_limit.size())
+                                                   if (!area_beyond_limit.empty())
                                                    {
                                                        // expand area_beyond_limit so that the inner hole fills in all the way back to the current layer's outline
                                                        // and use that to remove the regions in larger_area_below that should not use min XY because the regions are
                                                        // wide enough for a normal support to be placed there
                                                        larger_area_below = larger_area_below.difference(area_beyond_limit.offset(limit_distance + 10));
                                                    }
                                                }
+
+                                               {
+                                                   // Remove tiny "specs" in the `larger_area_below` polygon. As the polygon is a difference between the outline of
+                                                   // the current layer and the layer below this area is _usually_ a very narrow right surrounding the model (these
+                                                   // areas are desired). However, sometimes these areas become so narrow that only tiny dots are present. As the area
+                                                   // of both the rings and the specs are very small we cannot use the `removeSmallAreas` utility function. Instead, we
+                                                   // use the `removeSmallAreaCircumference` which removes polygons if both the area and circumference exceed a certain
+                                                   // threshold; we want a small circumference to differentiate between "rings" and "specs" (both have small area but
+                                                   // specs also have small circumference) and we want a small area to differentiate between larger blobs and "specs" in
+                                                   // polygon (both have a circumference but only "specs" have a "larger blobs" both have a small circumference but only
+                                                   // "specs" has a small area).
+                                                   constexpr size_t min_circumference = 500;
```

Comment:
Really nice improvement! One remark: can't we couple this min_circumference to the nozzle size? I would expect that any region with a circumference smaller than nozzle_diameter * pi can be removed. Similarly the min_area can be nozzle_diameter ** 2 / 4 * pi. That saves you two magic numbers.

Reply:
Good point, coupling these to the nozzle size does indeed make alot of sense.

But just "naively" coupling to a value from the settings in not always a good idea imo. Example being this line.
https://github.com/Ultimaker/CuraEngine/blob/e941d95fb717a652372f632225f1cdeeca632d4d/src/support.cpp#L895
Here we shrink one of the polygons to remove tiny areas. However to shrink this polygon we use 1/10th of the layer height. The layer height and polygon size have little to no correlation, and thus using layer height for the shrinkage also feels kind of like using magic numbers to me.

Reply:
https://github.com/Ultimaker/CuraEngine/pull/1803/commits/a94e60f1a10c4a5db30c5b8610464df39ab08eb6